### ISSUE TITLE
Commenting & Documentation

### ISSUE THREAD

**Issue by 3thancs** [3thancs/3thancs-intern-repo#46]
#### 🎯 Goal  
Learn when and how to write helpful comments and documentation.  

---

### ✅ Tasks  
- [x] **Research best practices for writing comments and documentation.**  
- [x] **Find an example of poorly commented code and rewrite the comments to be more useful.**  
- [x] **Write reflections in `clean_code.md`:**  
  - When should you add comments?  
  - When should you avoid comments and instead improve the code?  
- [x] **Commit and push your changes to GitHub.**  

---

### 🚀 Next Steps  
Move on to **Handling Errors & Edge Cases**.

**Comment by 3thancs** [comment]
**When should you add comments?**
* **Code can never tell the intent behind it**. For example, the code won't tell you that a 85/15 leads to overfitting.
* **Unavoidable logic** such as a complex algorithm such as Heterous Information Network 
 
 **When should you avoid comments and instead improve the code?**
* The code fails to tell the business logic of the code comments. For example, I create a constant named TAX_RATE = 0.3. The result code becomes Total Income * 0.3
* When comments are part of a function. For example, // step 1 clean data //step 6 print test result is redundant. Instead, it should be clean_data() and print(np.concatenate((y_pred.reshape(len(y_pred), 1), y_test.reshape(len(y_test),1)),1))
* Irrelevant and redundant variables such as d followed by comments explained as overdue_expenses. I renamed the variable to overdue_expenses instead.

* Find an example of poorly commented code and rewrite the comments to be more useful.
<details>
<summary>Click to see the poorly commented code</summary>
import pandas as pd
from sklearn.model_selection import train_test_split

# Load the data from csv
df = pd.read_csv('dataset.csv')

# Define x and y
X = df.drop('target', axis=1)
y = df['target']

# Split data into train and test
# Use 0.2 for test size
# Set random state to 42
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# Print the shapes
print(X_train.shape, X_test.shape)


<details>
<summary>Click to see the well commented code</summary>
# Import the libraries
import pandas as pd
from sklearn.model_selection import train_test_split

# Dataset contains cleaned features; target is a continuous variable
df = pd.read_csv('dataset.csv')
X = df.drop('target', axis=1)
y = df['target']

# We use a 80/20 split to ensure that no underfitting or overfitting occurs
# A polynomial regression algorithm was used due to a non-linear relationship between data points
# Random state = 42 ensures consistent results across several experiments
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# Verify split dimensions to ensure feature count consistency
print(X_train.shape, X_test.shape)

**Comment by FocusBearQA** [comment]
✅ **Approved**

🤖 Intern Review Bot: Great job completing this task! Here's a summary of my review:

- **Files Reviewed:**
  - Provided code examples and the markdown reflection section (`clean_code.md` reflection).

- **Code:**
  - Both poorly and well-commented code examples were relevant and show your understanding of the concept.
  - Comments in the 'well commented' section are context-aware and meaningful—good improvement.

- **Reflection:**
  - You answered both: when to add comments and when not to, with clear, specific examples.
  - Your reasoning went beyond reiterating the task; it showed solid understanding (e.g., mentioning intent, variable naming, and code readability over unnecessary comments).
  - You drew on your own analysis, rather than copying the task description, which shows real engagement.
  - Minor language improvements could be made (e.g., clarity in some sentences), but overall, it's easy to read and directly addresses the task.

- **AI-generated check:**
  - The submission has signs of original input and practical understanding, not just AI or copy-paste content.

- **No further action required.**
- **Next:** Go ahead and move on to "Handling Errors & Edge Cases" as suggested!

Keep up the good work and remember: clear, concise communication in code and docs goes a long way! 🚀

🤖 Intern Review Bot: Approved
